BangazonAPI/Controllers/PaymentTypeController.cs

```diff
@@ -0,0 +1,275 @@
+                            IsActive = reader.GetBoolean(reader.GetOrdinal("IsActive"))
+
+                        };
+
+
+
+                        paymentTypes.Add(paymentType);
+                    }
+
+
+                    reader.Close();
+
+                    return Ok(paymentTypes);
+                }
+            }
+        }
+
+
+
+        //GET: Code for getting a single paymentType (active or not)
+        [HttpGet("{id}", Name = "PaymentType")]
+        public async Task<IActionResult> GetSinglePaymentType([FromRoute] int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = $"SELECT Id, AcctNumber, [Name], CustomerId, IsActive from PaymentType WHERE Id=@id";
+                    cmd.Parameters.Add(new SqlParameter("@id", id));
+                    SqlDataReader reader = cmd.ExecuteReader();
+
+                    PaymentType paymentTypeToDisplay = null;
+
+                    while (reader.Read())
+                    {
+
+                        {
+                            paymentTypeToDisplay = new PaymentType
+                            {
+                                Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                                AcctNumber = reader.GetInt32(reader.GetOrdinal("AcctNumber")),
+                                Name = reader.GetString(reader.GetOrdinal("Name")),
+                                CustomerId = reader.GetInt32(reader.GetOrdinal("CustomerId")),
+                                IsActive = reader.GetBoolean(reader.GetOrdinal("IsActive"))
+
+                            };
+                        };
+                    };
+
+
+                    reader.Close();
+
+                    return Ok(paymentTypeToDisplay);
+                }
+            }
+        }
+
+        //  POST: Code for creating a paymentType
+        [HttpPost]
+        public async Task<IActionResult> PostPaymentType([FromBody] PaymentType paymentType)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"INSERT INTO PaymentType (AcctNumber, [Name], CustomerId, IsActive)
+                                                    OUTPUT INSERTED.Id
+                                                    VALUES (@AcctNumber, @Name, @CustomerId, @IsActive)";
+                    cmd.Parameters.Add(new SqlParameter("@AcctNumber", paymentType.AcctNumber));
+                    cmd.Parameters.Add(new SqlParameter("@Name", paymentType.Name));
+                    cmd.Parameters.Add(new SqlParameter("@CustomerId", paymentType.CustomerId));
+                    cmd.Parameters.Add(new SqlParameter("@IsActive", paymentType.IsActive));
```

Comment:
Rather than having the client POST an `isActive` boolean, let's just set it to `true` automatically when you create a new payment type. 

(When they first create it, `isActive` should never be false-- it should only be false if they've deleted it.)

Right now, the client has the ability to either POST a new payment type with `isActive` as false _or_, if they leave out the `isActive` property on the body of their POST request, it'll be set to false automatically. Too much power in the hands of the client! They can't handle it; they'll screw it up!

Reply:
I think it would default to true if they left it off (I set it up that way when I added it to the table in sequel), but it is now hard-coded as well.